debbuild

```diff
@@ -1574,8 +1722,24 @@ sub splitver {
      $req .= $ver if $ver;
   }
   # Pick up the details of versioned build requirements
-  my ($pkg,$rel,$ver) = $req =~ /([\w.+-]+)\s*([><=]+)\s*([\w:.~+-]+)/;
+  my ($pkg,$rel,$ver) = $req =~ /([\w.+\-()]+)\s*([><=]+)\s*([\w:.~+-]+)/;
+  $pkg =~ s/\(([^\)]+)\)/-$1/g;
   $rel =~ s/^([><])$/$1$1/;
+
+  # We need this workaround as _ is foridden in the package name
+  #  and probably was changed to - on building such packages.
+  $pkg =~ tr/_/-/;
+
+  # Fix strict version checking by including release and epoch in case of specifying the packages currently building
+  if ( $rel eq '=' ) {
+    foreach my $p ( keys(%pkgdata) ) {
+      if ( $ver eq $pkgdata{main}{version} ) {
+        $ver = (defined $pkgdata{main}{epoch} ? "$pkgdata{main}{epoch}:" : '').$ver."-".$pkgdata{main}{release};
+        last;
+      }
+    }
+  }
```

Comment:
I don't want this hack in debbuild. This is a bug in your spec file and you need to fix it.

Reply:
This changes was introduced due to the difference in processing version in spec and in dpkg way.
`spec` is checking only version number if `=` is used with the version only, while dpkg is checking EVR with strict checking with `=`. The other option is using `~` like: `>= 1.2.3-~`.

Reply:
Yes, but openSUSE's usage of `foo = %version` is almost universally wrong too. I don't want to enable even more bad practices by including that.

Reply:
Why it's bad if we only need to ensure the right version of the package, but not the release? And it's totally bad idea to rewrite all the specs just because of it. It's just different ways RPM and DPKG is checking the version. For RPM the version is the only version part, but not whole EVR value as for DPKG. Not sure if it's a good idea to push everyone to rewete the specs because of it.

Reply:
It's bad because OBS' auto-rebuilding behavior makes it so that it's possible for ABI mismatches to be exist without an expression to force them to be fixed. Also, if you add patches (which bump release but not version), you can create subtle conflicts that break at runtime without being correctly expressed as RPM dependencies. This happens often enough for me in openSUSE that pretty much every package I touch that has this problem gets adjusted to use `%version-%release` instead.

Reply:
Unfortunatelly the word we live in is not ideal, and there are a lot of the packages already building with checking for the version only in `specs`. Without such hacks we will just produce broken packages which can't be installed.

Reply:
So the difference here is that I think you can fix your own packages that you're shipping. You're rebuilding them from openSUSE Factory for Debian/Ubuntu, which means you _already_ have the sources and the ability to make SRs to fix things.

Reply:
Sure it can be fixed in the spec, but the goal of our project is to avoid spec modifications as much as possible to decrease the number of entties to maintain, as each additional spec change can cause package brokage on changing original spec file where the package was branched from and each time we have to fix spec files on such modifications, but such change for original spec makes no sense as for rpm it's normal.

Reply:
Except for the case where you're doing this, it's also bad for RPM environments. It works entirely coincidentally but is a bad behavior for all the reasons I said earlier.

Reply:
It's clear what you mean, but for us it's impossible to push maintainers for 20+ packages with such changes.

Reply:
And I've removed it already from the commit